genesyscloud/resource_genesyscloud_architect_ivr.go

```diff
@@ -0,0 +1,247 @@
+	scheduleGroupId := buildSdkDomainEntityRef(d, "schedule_group_id")
+
+	sdkConfig := meta.(*providerMeta).ClientConfig
+	architectApi := platformclientv2.NewArchitectApiWithConfig(sdkConfig)
+
+	ivrBody := platformclientv2.Ivr{
+		Name:             &name,
+		Description:      &description,
+		Dnis:             buildSdkIvrDnis(d),
+		OpenHoursFlow:    openHoursFlowId,
+		ClosedHoursFlow:  closedHoursFlowId,
+		HolidayHoursFlow: holidayHoursFlowId,
+		ScheduleGroup:    scheduleGroupId,
+	}
+
+	log.Printf("Creating IVR config %s", name)
+	ivrConfig, _, err := architectApi.PostArchitectIvrs(ivrBody)
+	if err != nil {
+		return diag.Errorf("Failed to create IVR config %s: %s", name, err)
+	}
+
+	d.SetId(*ivrConfig.Id)
+
+	log.Printf("Created IVR config %s %s", name, *ivrConfig.Id)
+	return readIvrConfig(ctx, d, meta)
+}
+
+func readIvrConfig(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	sdkConfig := meta.(*providerMeta).ClientConfig
+	architectApi := platformclientv2.NewArchitectApiWithConfig(sdkConfig)
+
+	log.Printf("Reading IVR config %s", d.Id())
+	ivrConfig, resp, getErr := architectApi.GetArchitectIvr(d.Id())
+	if getErr != nil {
+		if resp != nil && resp.StatusCode == 404 {
+			d.SetId("")
+			return nil
+		}
+		return diag.Errorf("Failed to read IVR config %s: %s", d.Id(), getErr)
+	}
+
+	if *ivrConfig.State == "deleted" {
+		d.SetId("")
+		return nil
+	}
+
+	d.Set("name", *ivrConfig.Name)
+	d.Set("dnis", flattenIvrDnis(ivrConfig.Dnis))
+
+	if ivrConfig.Description != nil {
+		d.Set("description", *ivrConfig.Description)
+	} else {
+		d.Set("description", nil)
+	}
+
+	if ivrConfig.OpenHoursFlow != nil {
+		d.Set("open_hours_flow_id", *ivrConfig.OpenHoursFlow.Id)
+	} else {
+		d.Set("open_hours_flow_id", nil)
+	}
+
+	if ivrConfig.ClosedHoursFlow != nil {
+		d.Set("closed_hours_flow_id", *ivrConfig.ClosedHoursFlow.Id)
+	} else {
+		d.Set("closed_hours_flow_id", nil)
+	}
+
+	if ivrConfig.HolidayHoursFlow != nil {
+		d.Set("holiday_hours_flow_id", *ivrConfig.HolidayHoursFlow.Id)
+	} else {
+		d.Set("holiday_hours_flow_id", nil)
+	}
+
+	if ivrConfig.ScheduleGroup != nil {
+		d.Set("schedule_group_id", *ivrConfig.ScheduleGroup.Id)
+	} else {
+		d.Set("schedule_group_id", nil)
+	}
+
+	log.Printf("Read IVR config %s %s", d.Id(), *ivrConfig.Name)
+	return nil
+}
+
+func updateIvrConfig(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	name := d.Get("name").(string)
+	description := d.Get("description").(string)
+	openHoursFlowId := buildSdkDomainEntityRef(d, "open_hours_flow_id")
+	closedHoursFlowId := buildSdkDomainEntityRef(d, "closed_hours_flow_id")
+	holidayHoursFlowId := buildSdkDomainEntityRef(d, "holiday_hours_flow_id")
+	scheduleGroupId := buildSdkDomainEntityRef(d, "schedule_group_id")
+
+	sdkConfig := meta.(*providerMeta).ClientConfig
+	architectApi := platformclientv2.NewArchitectApiWithConfig(sdkConfig)
+
+	ivrBody := platformclientv2.Ivr{
```

Comment:
You'll need to fetch the latest version when doing an update. See https://github.com/MyPureCloud/terraform-provider-genesyscloud/blob/main/genesyscloud/resource_genesyscloud_group.go#L235 for the pattern

Reply:
Oh thanks for the catch. Looks like the API doesn't require version (unlike groups and users) when updating, so I forgot there is a version property. Added.